### PR TITLE
Add new attribute 'unit' to 'bus' component

### PR DESCRIPTION
**Pull request by euronion**
## Changes proposed in this Pull Request

Add a `unit` attribute to the `bus` component to explicitly describe the unit of the bus.
Purely descriptive at the moment.
Useful for non-electricity buses (e.g. hydrogen stores in `kg` or `t`).

It clashes (softly) with other documented attributes, like the `marginal_price` ("in currency/MWh")

## Checklist

- [x] ~Code changes are sufficiently documented; i.e. new functions contain docstrings and further explanations may be given in `doc`.~
- [x] ~Unit tests for new features were added (if applicable).~
- [x] ~Newly introduced dependencies are added to `environment.yaml`, `environment_docs.yaml` and `setup.py` (if applicable).~
- [x] ~A note for the release notes `doc/release_notes.rst` of the upcoming release is included.~

### CODE REVIEW

**Review by fneum — COMMENTED** [review]
Three thoughts:

- default of the unit attribute should be MW not None
- Is it a problem that this mixes units of power (MW) and units of energy (kg)? Wouldn't a flow kg/h be more consistent? The marginal price unit would then be EUR/kg.
- The documentation can stay on MW/MWh elsewhere. That's fine.

**Comment by fneum** [comment]
Should the unit of a bus be a unit of energy or a unit of power?

**Comment by euronion** [comment]
Sorry for not following up on this earlier.

The unit should be a unit of power (energy/time ratio) to preserve consistency between components as the time dimension is represented and by snapshots, thus determining the flow magnitude.

File: pypsa/component_attrs/buses.csv [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
-v_ang,series,radians,0.,Voltage angle,Output
-marginal_price,series,currency/MWh,0.,Locational marginal price from LOPF from power balance constraint,Output
+"attribute","type","unit","default","description","status",
+"name","string","n/a","n/a","Unique name","Input (required)",
+"v_nom","float","kV",1,"Nominal voltage","Input (optional)",
+"type","string","n/a","n/a","Placeholder for bus type. Not yet implemented.","Input (optional)",
+"x","float","n/a",0,"Position (e.g. longitude)"," the Spatial Reference System Identifier (SRID) is set in network.srid.","Input (optional)"
+"y","float","n/a",0,"Position (e.g. latitude)"," the Spatial Reference System Identifier (SRID) is set in network.srid.","Input (optional)"
+"carrier","string","n/a","AC","Energy carrier: can be ""AC"" or ""DC"" for electrical buses, or ""heat"" or ""gas"".","Input (optional)",
+"unit","string","n/a","None","Unit of the bus’ carrier if the implicitly assumed unit (""MW"") is inappropriate (e.g. ""t"", ""MWh_th""). Only descriptive. Does not influence any PyPSA functions.","Input (optional)",
```
Comment (fneum) [inline pypsa/component_attrs/buses.csv:8]:
```suggestion
"unit","string","n/a","None","Unit of the bus’ carrier if the implicitly assumed unit (""MW"") is inappropriate (e.g. ""t/h"", ""MWh_th/h""). Only descriptive. Does not influence any PyPSA functions.","Input (optional)",
```

**Pull request merged**